### PR TITLE
Setup CI/CD for business registry.

### PR DESCRIPTION
**Pull request by pwei1018**
https://github.com/bcgov/entity/issues/22772

*Description of changes:*
1. CI flow;
2. CD flow;
3. preview site;
4. codecov configuration;

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-ar-dev--pr-10-v0rb7vdx.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-registry-dev--pr-10-2iyepwf7.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-registry-dev--pr-10-2iyepwf7.web.app

File: codecov.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+comment:
+  layout: "diff,flags,tree"
+  behavior: default
+  require_changes: true
+
+flags:
+  businessregistryui:
+    paths:
+      - web/site
```
Comment (deetz99) [inline codecov.yaml:37]: Should this still be web/site?

Comment (pwei1018) [reply codecov.yaml:37]: Fixed

**Review by deetz99 — APPROVED** [review]

**Review by thorwolpert — APPROVED** [review]

**Pull request merged**